### DISCUSSION TITLE
How to work with multiple schemas in PostgreSQL

### DISCUSSION THREAD

**Discussion by oovm** [SeaQL/sea-orm#2308] [Q&A]
Our database has multiple schemas, and some tables' foreign key constraints exist across schemas.

The generated entries referencing `super::table::entry` will show that such table name cannot be found.

Is there some generation methods applicable to multiple schemas?

I think path can be:

```yaml
src/
  - schema1/
    - table1.rs
    - table2.rs
    - mod.rs
  - schema2/
    - table1.rs
    - table2.rs
    - mod.rs
   lib.rs
   prelude.rs
```

---

P.S. I am a backend developer, not a DBA, and I have no right to decide how to design the database.

**Comment by sjunepark** [comment]
For me, I'm using the workaround below.
(I have 2 schemas `dart` and `smes`)

```shell
sea-orm-cli generate entity -o crates/db/src/entities/dart -s dart
sea-orm-cli generate entity -o crates/db/src/entities/smes -s smes
```

and create `crates/db/src/entities/mod.rs` with the code below
```rust
pub mod dart;
pub mod smes;
```

**Comment by oovm** [comment]
In normal cases, it works

But if there are foreign keys and type definitions, an error will be reported that the type is missing.

**Reply by Cikmo** [reply]
Ran into this today. sjunepark's solution is what I came up with, but relations don't really work correctly. My workaround for now is just editing the mod.rs file for the schema to import the needed to entity from the other schema. The annoying part is I need to manually do this every time I generate the schemas, as it overwrites my changes.
